### PR TITLE
BNGP-2809: Main Credits Journey - Metric File Upload

### PR DESCRIPTION
**Pull request by usman-j-shaikh**

### CODE REVIEW

File: packages/webapp/src/routes/credits-purchase/credits-development-location.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  post: async (request, h) => {
+    const confirmDevDetails = request.payload.confirmDevDetails
+    const metricUploadLocation = request.yar.get(constants.redisKeys.CREDITS_METRIC_LOCATION)
+    request.yar.set(constants.redisKeys.METRIC_FILE_CHECKED, confirmDevDetails)
+    if (confirmDevDetails === constants.creditsCheckDetails.NO) {
+      await deleteBlobFromContainers(metricUploadLocation)
+      request.yar.clear(constants.redisKeys.CREDITS_METRIC_LOCATION)
+      return h.redirect(constants.routes.CREDITS_UPLOAD_METRIC)
+    } else if (confirmDevDetails === constants.creditsCheckDetails.YES) {
+      return h.redirect('#')
```
Comment (KiranVarma02) [inline packages/webapp/src/routes/credits-purchase/credits-development-location.js:20]: is this redirect correct?

File: packages/webapp/src/routes/credits-purchase/credits-metric-upload.js [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+            text: constants.uploadErrors.uploadFailure,
+            href: UPLOAD_CREDIT_METRIC_ID
+          }]
+        })
+      }
+  }
+}
+
+const handlers = {
+  onGet: async (_request, h) => h.view(constants.views.CREDITS_UPLOAD_METRIC),
```
Comment (KiranVarma02) [inline packages/webapp/src/routes/credits-purchase/credits-metric-upload.js:78]: could we please follow same conventions as other route files in the PR

File: packages/webapp/src/routes/credits-purchase/credits-development-location.js (hunk 1 above)
Comment (AjinkyaJawalkar) [reply packages/webapp/src/routes/credits-purchase/credits-development-location.js:20]: Further page is not developed and when this screen was developed at that time complete prototype was not available so added redirecting to "#"

Comment (AjinkyaJawalkar) [reply packages/webapp/src/routes/credits-purchase/credits-development-location.js:20]: Could you please check the following change made? 

https://github.com/DEFRA/biodiversity-net-gain-service/pull/618/commits/4eba8a811e1773828a18cb4f44e8f5a185911031

Comment (AjinkyaJawalkar) [reply packages/webapp/src/routes/credits-purchase/credits-development-location.js:20]: Yes. The actual page is not developed yet and when this PR was created any further prototype pages was not there

**Pull request closed without merging**